pkg/registration/spoke/bootstrapkubeconfigsmanager/bootstrapkubeconfiginuse.go

```diff
@@ -0,0 +1,53 @@
+package bootstrapkubeconfigsmanager
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// bootstrapKubeConfigInUse is the registration spoke's current in used bootstrap kubeconfig.
+type bootstrapKubeConfigInUse interface {
+	// KubeConfigData returns the kubeconfig data of the bootstrap kubeconfig and the credentials currently in use.
+	KubeConfigData() (map[string][]byte, error)
+
+	// Update updates the kubeconfig data of the bootstrap kubeconfig in use.
+	Update(ctx context.Context, kubeconfigData map[string][]byte) error
+}
+
+var _ bootstrapKubeConfigInUse = &bootstrapKubeConfigInUseImpl{}
+
+type bootstrapKubeConfigInUseImpl struct {
+	secretName      string
+	secretNamespace string
```

Comment:
remove secretName and use from constant pkg.

Reply:
@qiujian16 

From the code we know, the spoke compoennt namespace is not always the  defaultSpokeComponentNamespace, it depends on which ns the agent is installed, which means it's neccessary to keep the `secretNamespace`.

```
// NewAgentOptions returns the flags with default value set
func NewAgentOptions() *AgentOptions {
	opts := &AgentOptions{
		HubKubeconfigDir:   "/spoke/hub-kubeconfig",
		ComponentNamespace: defaultSpokeComponentNamespace,
		CommoOpts:          NewOptions(),
	}
	// get component namespace of spoke agent
	nsBytes, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
	if err == nil {
		opts.ComponentNamespace = string(nsBytes)
	}
	return opts
}
```